.github/workflows/ci-schedule.yml

```diff
@@ -38,9 +38,9 @@ jobs:
           # We need to guess version via git tags.
           fetch-depth: 0
       - name: install Go
-        uses: actions/setup-go@v3
+        uses: actions/setup-go@v5
         with:
-          go-version: 1.21.8
+          go-version-file: .go-version
```

Comment:
It seems the `go-version-file` accepts the path of `go.mod` or `go.work`.
The `actions/setup-go` first tries to parse the go version from `go.mod` or `go.work` by a regular expression `/^go (\d+(\.\d+)*)/m`, and fall back to `trim` the specified file, echo the [relevant code](https://github.com/actions/setup-go/blob/be1aa1186e83689f3e45adaf3697af86f816fb2e/dist/setup/index.js#L88328-L88336) here:
```js
function parseGoVersionFile(versionFilePath) {
    const contents = fs_1.default.readFileSync(versionFilePath).toString();
    if (path.basename(versionFilePath) === 'go.mod' ||
        path.basename(versionFilePath) === 'go.work') {
        const match = contents.match(/^go (\d+(\.\d+)*)/m);
        return match ? match[1] : '';
    }
    return contents.trim();
}
```
That would work I think.

Reply:
yeah that could work ... never saw go.work being used

Reply:
Yes, Karmada-io/karmada only has one module inside, so it doesn't need the go.work.